std/typecons.d

```diff
@@ -3675,6 +3675,59 @@ struct Nullable(T)
 
     private bool _isNull = true;
 
+    /**
+     * Compares two Nullable values.
+     * - If one is null and the other is not, the null one is considered smaller.
+     * - If both are null, they are equal.
+     * - If both are non-null, compares the payloads.
+     *
+     * Returns:
+     *     Negative if `this < rhs`, zero if equal, positive if `this > rhs`.
+     */
+    int opCmp(Rhs)(auto ref Rhs rhs) const
+    if (is(typeof(_value.payload < rhs.get)) && is(typeof(_value.payload > rhs.get)))
```

Comment:
This test will return `true` for *any* type with a `get` method that returns a value comparable to the payload, even if it is not a `Nullable` type. The correct way to check whether `Rhs` is a `Nullable` type whose payload is comparable to this `Nullable`'s payload is like this:
```d
    if (is(Rhs == Nullable!U, U) && is(typeof(this.get < rhs.get)))
```
(Strictly speaking, it doesn't matter whether you use `payload` or `get` in the `typeof(...)` expression, but you should use the same on both sides for consistency.)